pageserver/src/tenant/timeline/layer_manager.rs

```diff
@@ -238,14 +238,14 @@ impl LayerManager {
     /// Called when garbage collect the timeline. Returns a guard that will apply the updates to the layer map.
     pub(crate) fn finish_gc_timeline(
         &mut self,
-        layer_removal_cs: Arc<tokio::sync::OwnedMutexGuard<()>>,
+        layer_removal_cs: &Arc<tokio::sync::OwnedMutexGuard<()>>,
         gc_layers: Vec<Arc<dyn PersistentLayer>>,
         metrics: &TimelineMetrics,
     ) -> Result<ApplyGcResultGuard> {
         let mut updates = self.layer_map.batch_update();
         for doomed_layer in gc_layers {
             Self::delete_historic_layer(
-                layer_removal_cs.clone(),
+                layer_removal_cs,
```

Comment:
All the changes related to this `Arc` => `&Arc` feel out of tune with the rest of this PR.

Reply:
They are unrelated cleanups yes.

I added them here because of the review latency, but ok, let's split them.